tests/functional/t/typevar_naming_style_default.py

```diff
@@ -0,0 +1,53 @@
+"""Test case for typevar-name-incorrect-variance with default settings"""
+# pylint: disable=too-few-public-methods
+
+from typing import TypeVar
+
+
+class MyClass:
+    """Class to be used to bound TypeVar's"""
+
+
+# PascalCase names with prefix
+GoodNameT = TypeVar("GoodNameT")
+_T = TypeVar("_T")
+_GoodNameT = TypeVar("_GoodNameT")
+__GoodNameT = TypeVar("__GoodNameT")
+GoodNameWithoutContra = TypeVar(  # [typevar-name-incorrect-variance]
+    "GoodNameWithoutContra", contravariant=True
+)
+GoodNameT_co = TypeVar("GoodNameT_co", covariant=True)
+GoodNameT_contra = TypeVar("GoodNameT_contra", contravariant=True)
+GoodBoundNameT = TypeVar("GoodBoundNameT", bound=MyClass)
+GoodBoundNameT_co = TypeVar(  # [typevar-name-incorrect-variance]
+    "GoodBoundNameT_co", bound=MyClass
+)
+GoodBoundNameT_contra = TypeVar(  # [typevar-name-incorrect-variance]
+    "GoodBoundNameT_contra", bound=MyClass
+)
+
+# PascalCase names without prefix
+AnyStr = TypeVar("AnyStr")
+BadNameWithoutContra = TypeVar(  # [typevar-name-incorrect-variance]
+    "BadNameWithoutContra", contravariant=True
+)
+BadName_co = TypeVar("BadName_co", covariant=True)
+BadName_contra = TypeVar("BadName_contra", contravariant=True)
+
+# camelCase names with prefix
+T_goodName = TypeVar("T_goodName")  # [invalid-name]
+T_goodNameWithoutContra = TypeVar(  # [invalid-name, typevar-name-incorrect-variance]]
+    "T_goodNameWithoutContra", contravariant=True
+)
+T_goodName_co = TypeVar("T_goodName_co", covariant=True)  # [invalid-name]
+T_goodName_contra = TypeVar("T_goodName_contra", contravariant=True)  # [invalid-name]
+
+# PascalCase names with lower letter prefix in tuple assignment
+(
+    a_BadName,  # [invalid-name]
+    a_BadNameWithoutContra,  # [invalid-name, typevar-name-incorrect-variance]
+) = TypeVar("a_BadName"), TypeVar("a_BadNameWithoutContra", contravariant=True)
+GoodName_co, a_BadName_contra = TypeVar(  # [invalid-name]
+    "GoodName_co", covariant=True
+), TypeVar("a_BadName_contra", contravariant=True)
+GoodName_co, VAR = TypeVar("GoodName_co", covariant=True), "a string"
```

Comment:
Maybe at the end. Could you add `CALLABLE_T`, `DeviceTypeT`, and `DeviceType`? The last just so we actual test that part of the regex and know that we change the right thing should we do so.

Reply:
https://github.com/PyCQA/pylint/pull/5221/commits/cebeeac01b1245a65740006a0af650ebc23005a3